backend/lucene/src/main/java/org/hibernate/search/backend/lucene/orchestration/impl/LuceneParallelWorkOrchestratorImpl.java

```diff
@@ -94,10 +94,14 @@ public void run() {
 				result.complete( work.execute( context ) );
 			}
 			catch (Throwable e) {
-				context.getIndexAccessor().cleanUpAfterFailure( e, work.getInfo() );
-				result.completeExceptionally( e );
+				markAsFailed( e );
 			}
 		}
+
+		public void markAsFailed(Throwable throwable) {
+			context.getIndexAccessor().cleanUpAfterFailure( throwable, work.getInfo() );
```

Comment:
This doesn't seem necessary if we just failed to *submit* the work. If that happens we didn't even touch the index accessor.